### DISCUSSION TITLE
Apps restart when Xdrip+ refreshes BG value

### DISCUSSION THREAD

**Discussion by CountScotula** [Navid200/xDrip#150] [Dexcom]
I'm seeing an issue where apps restart when Xdrip+ updates the BG value. The behavior I've noticed is subtle, but frustrating as heck.

A typical situation might be that I'm composing a message in a web form in Chrome and then all of a sudden, the page seems to refresh and my draft is gone. I check Xdrip+ and it has just refreshed. 

I see similar things in social media apps like Facebook or Instagram where I'm scrolling along and then instantly I'm back at the top of the feed. These *always* coincide with Xdrip+ refreshes.

I had been having the bluetooth headset dropped stream issue others reported with Android 12 and I thought these behaviors were related. But after fiddling with settings and apps and updating Xdrip+, the bluetooth issue has resolved. The restarting apps issue remains, however.

I'm currently using a Pixel 6 Pro running Android 12 (patch level Nov 5) and Xdrip+ nightly build from 2nd Dec 2021. I'm using  the 640G/EverSense hardware data source to collect BG from the locally installed Dexcom G6 app. But I've also tried switching to the Dexcom Share source and still see the behavior.

I'm hoping this might be something I can solve in settings, since I haven't seen anyone else reporting this exact issue.

**Comment by Navid200** [comment]
The only time I have seen an interruption caused by xDrip while having proper connectivity and performing proper readings is this:
https://github.com/NightscoutFoundation/xDrip/issues/1859
But, that was really caused by Roku.

I am going to ask others, with more experience, to have a look.  
Thanks for posting.

**Reply by CountScotula** [reply]
Thanks. Yes, I was following that because I was seeing the bluetooth audio issue. And I uninstalled the Roku app at that time. FWIW, I did not see improvement in that issue at that point, though I'm not seeing bluetooth issues now. And I still don't have the Roku app installed.

**Comment by Navid200** [comment]
@CountScotula 
Would you please open an issue?  https://github.com/NightscoutFoundation/xDrip/issues
Please include a link to this thread.

**Reply by CountScotula** [reply]
You bet. Thanks for helping me sort this. I really appreciate it.

**Comment by Navid200** [comment]
What would help is if you put yourself in our shoes.
There are others who are using that same phone and are not experiencing this.

I want to help get an answer for the problem you are experiencing.  And I have made suggestions.
You can entertain them.
Or, you can wait for other suggestions.
please look at the existing open issues.  Some of them have been there for years.

I'm trying to help.
There are hundreds of others issues.  And they all need attention.

**Reply by CountScotula** [reply]
My apologies. I haven't meant to imply that I'm demanding attention. I only asked a question about this issue on Facebook to see if others using the app had experienced it. I never even intended to report it as an issue. I've only done so at your suggestion and because I thought people working on the app might like to know if there is a deeper issue.

I do appreciate your suggestions, but given the amount of basic troubleshooting I've done on my own, I'll wait for more concrete suggestions.
